src/internal/operators/repeatWhen.ts

```diff
@@ -38,96 +38,94 @@ import { SimpleOuterSubscriber, innerSubscribe, SimpleInnerSubscriber } from '..
  * @name repeatWhen
  */
 export function repeatWhen<T>(notifier: (notifications: Observable<void>) => Observable<any>): MonoTypeOperatorFunction<T> {
-  return (source: Observable<T>) => lift(source, function (this: Subscriber<T>, source: Observable<T>) {
-    const subscriber = this;
-    const subscription = new Subscription();
-    let innerSub: Subscription | null;
-    let syncResub = false;
-    let completions$: Subject<void>;
-    let isNotifierComplete = false;
-    let isMainComplete = false;
+  return (source: Observable<T>) =>
+    lift(source, function (this: Subscriber<T>, source: Observable<T>) {
+      const subscriber = this;
+      const subscription = new Subscription();
```

Comment:
issue: It's not clear to me why this `Subscription` instance is necessary. I mean, as it's returned, it's added to the subscriber - within `Observable#subscribe` - but the subscriber is the destination for both `OperatorSubscribers` in this implementation. If there's a reason for this - given that this is so unlike each of the other operators - I think it warrants a comment.

Reply:
Ah... yeah. I didn't actually touch this one much other than to change it to use `OperatorSubscriber`... but I can rework it a little to see what I can do there.

Reply:
Updated.